File: CaseStudy/stock/src/main/java/com/stores/stock/model/BilledProduct.java
package com.stores.stock.model;

import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;

@Entity
@Table(name="BillingProduct")
public class BilledProduct {
    @Id
    private int billId;
    @Id
    private int productId;
    private int cost;
    private int quantity;

    public int getBillId() {
        return billId;
    }

    public void setBillId(int billId) {
        this.billId = billId;
    }

    public int getProductId() {
        return productId;
    }

    public void setProductId(int productId) {
        this.productId = productId;
    }

    public int getCost() {
        return cost;
    }

    public void setCost(int cost) {
        this.cost = cost;
    }

    public int getQuantity() {
        return quantity;
    }

    public void setQuantity(int quantity) {
        this.quantity = quantity;
    }

    public BilledProduct(int billId, int productId, int cost, int quantity) {
        this.billId = billId;
        this.productId = productId;
        this.cost = cost;
        this.quantity = quantity;
    }

    public BilledProduct() {
    }

    @Override
    public String toString() {
        return "BilledProduct{" +
                "billId=" + billId +
                ", productId=" + productId +
                ", cost=" + cost +
                ", quantity=" + quantity +
                '}';
    }
}

File: CaseStudyKotlin/demo/src/main/kotlin/com/store/demo/model/BilledProduct.kt
package com.store.demo.model

import com.fasterxml.jackson.annotation.JsonIgnore
import java.io.Serializable
import java.util.*
import javax.persistence.Entity
import javax.persistence.Id
import javax.persistence.IdClass
import javax.persistence.Table
@Entity
@Table(name = "BillingProduct")
@IdClass(BilledProduct::class)
class BilledProduct : Serializable {
    @Id 
    @JsonIgnore
    var billId = 0
    @Id
    var productId = 0
    var cost = 0
    var quantity = 0

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o !is BilledProduct) return false
        val that = o
        return billId == that.billId && productId == that.productId && cost == that.cost && quantity == that.quantity
    }

    override fun hashCode(): Int {
        return Objects.hash(billId, productId, cost, quantity)
    }

    constructor(billId: Int, productId: Int, cost: Int, quantity: Int) {
        this.billId = billId
        this.productId = productId
        this.cost = cost
        this.quantity = quantity
    }

    constructor() {}

    override fun toString(): String {
        return "BilledProduct{" +
                "billId=" + billId +
                ", productId=" + productId +
                ", cost=" + cost +
                ", quantity=" + quantity +
                '}'
    }
}